src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift

```diff
@@ -393,6 +441,11 @@ class CoronaTestService {
 			return
 		}
 
+		guard coronaTest.finalTestResultReceivedDate == nil else {
+			completion(.success(coronaTest.testResult))
+			return
+		}
```

Comment:
Sadly we cannot do this, because according to the tech spec [the result shall also be updated even if a result other than pending has been received for the same test previously (and regardless of whether the result is stored or not).](https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/rapid-test/docs/spec/corona-test-client.md#poll-the-test-result-in-foreground) This is only stated for the foreground, but whether this is also valid for the background is [still open](https://github.com/corona-warn-app/cwa-app-tech-spec/pull/37#discussion_r606218513). Maybe we should leave this check out for now? 🤔

Reply:
Yes, I think the background task should call the update only if the test result does not exist. Because otherwise it would result in multiple notifications. Therefore I have added a "force" flag for the update.

Reply:
@nickguendling could you have a look, if its OK now?